src/Hooks/StrictEqualityHooks.php

```diff
@@ -109,15 +122,28 @@ private static function isEqualOrdered(Atomic $first_type, Atomic $second_type):
             return true;
         }
 
-        if ($first_type instanceof Atomic\TArray && $second_type instanceof Atomic\TArray) {
+        if ($first_type instanceof Atomic\TList && $second_type instanceof Atomic\TList) {
+            return true;
+        }
+
+        if ($first_type instanceof Atomic\TIterable || $first_type instanceof Atomic\TKeyedArray || $first_type instanceof Atomic\TArray) {
+            return false;
+        }
+
+        if ($first_type instanceof Atomic\TNamedObject && $second_type instanceof Atomic\TNamedObject) {
+            return true;
+        }
+
+        if ($first_type instanceof Atomic\TObject && $second_type instanceof Atomic\TNamedObject) {
             return true;
         }
 
-        if ($first_type instanceof Atomic\TKeyedArray && $second_type instanceof Atomic\TKeyedArray) {
+        if ($first_type instanceof Atomic\TObject && $second_type instanceof Atomic\TObject) {
             return true;
         }
 
-        if ($first_type instanceof Atomic\TKeyedArray && $second_type instanceof Atomic\TArray) {
+        // generic same or parent class
+        if ($first_type instanceof $second_type) {
```

Comment:
If you want to add this here, we should handle every case we know it's not true before. For example, you must return false on KeyedArray comparisons or else you'll consider a valid change